src/main/kotlin/com/exactpro/th2/sim/template/rule/KotlinFIXRule.kt

```diff
@@ -139,7 +141,7 @@ class KotlinFIXRule(field: Map<String, Any?>) : MessageCompareRule() {
                     .addField(FixFields.EXEC_ID, execId.incrementAndGet())
                     .with(sessionAlias = aliasdc1)
             )
-        } else {
+        } else synchronized(lock) {
             sellOrdersAndIds.add(OrderWithId(incomeMessage, incomeOrderId))
```

Comment:
This block should be executed inside single synchronized block because we need atomically read all three consequent orders from queues for one test case (after making sure that they are there). Adding new order don't need to be in the same block.